leakcanary-deobfuscation-gradle-plugin/src/main/java/com/squareup/leakcanary/deobfuscation/CopyObfuscationMappingFileTask.kt

```diff
@@ -0,0 +1,56 @@
+package com.squareup.leakcanary.deobfuscation
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+@CacheableTask
+open class CopyObfuscationMappingFileTask : DefaultTask() {
+
+  @Input
+  @SkipWhenEmpty
+  @PathSensitive(PathSensitivity.RELATIVE)
+  lateinit var mappingFile: File
+
+  @Input
+  @SkipWhenEmpty
+  @PathSensitive(PathSensitivity.RELATIVE)
+  lateinit var mergeAssetsDirectory: File
+
+  @get:OutputFile
+  @get:PathSensitive(PathSensitivity.RELATIVE)
+  val leakCanaryAssetsOutputFile: File
+    get() = File(mergeAssetsDirectory, "leakCanaryObfuscationMapping.txt")
```

Comment:
Just wondering how we could share this file constant with the `HeapAnalyzerService` that also defines it has a constant.
It would either put the plugin inside the leak canary analyzer as a dependency or the heap analyzer in the classpath during the build. Maybe the later is not bad actually: -> we could add a dependency from the plugin to the `HeapAnalyzerService` and use this constant.

Reply:
So you're suggesting that we should add
`implementation project(':leakcanary-deobfuscation-gradle-plugin')`
to `shark` module that contains `HeapAnalyzerService` so we can have a single constant with file mapping file name?
`shark` module is a dependency of almost all other modules so that means that plugin would become a dependency of them too. Maybe we could do it the other way around - add `implementation project(':shark')` to the plugin and have the constant defined in the `HeapAnalyzerService`?

Reply:
yes, we could also do that. Or, as any other design problem, we can introduce a new entity to add a layer of indirection -> add a new shared module between shark and the plugin... ?

Reply:
@pyricau what do you think about this?

Reply:
I would avoid adding the gradle plugin as a dependency to the runtime artifacts. Similarly wouldn't add the runtime analysis code to the plugin.

If you want to share the constant, create a new small artifact with just that constant and add it as a dependency to the two projects that use it.

That being said, I also think it's totally ok to have an implicit contract with regards to the file name. We might still want to change the file name and behavior in the  future, so making it a public constant makes it hard to change.

Reply:
Makes sense, I'm leaving it as is for now. Thank you.